pkg/cnab/directory_parameter.go

```diff
@@ -0,0 +1,104 @@
+package cnab
+
+import (
+	"encoding/json"
+
+	"github.com/cnabio/cnab-go/bundle/definition"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/pkg/errors"
+)
+
+const (
+	DirectoryExtensionShortHand    = "directory-parameter"
+	DirectoryParameterExtensionKey = PorterExtensionsPrefix + DirectoryExtensionShortHand
+)
+
+// DirectoryParameterDefinition represents those parameter options
+// That apply exclusively to the directory parameter type
+type DirectoryParameterDefinition struct {
+	Writeable bool `yaml:"writeable,omitempty"`
+	// UID and GID should be ints, however 0 is the default value for int type
+	// But is also a realistic value for UID/GID thus we need to make the type interface
+	// To detect the case that the values weren't set
+	GID interface{} `yaml:"gid,omitempty" json:"gid,omitempty"`
+	UID interface{} `yaml:"uid,omitempty" json:"uid,omitempty"`
+}
+
+// MountParameterSource represents a parameter using a docker mount
+// As a its source with the provided options
+type MountParameterSourceDefn struct {
+	mount.Mount `yaml:",inline"`
```

Comment:
Okay I can use a string. My idea with using a larger object was so that we didn't have to parse the string, but it should otherwise simplify things a bit so I'll make the change.